src/torchjd/autojac/_transform/aggregate.py

```diff
@@ -0,0 +1,165 @@
+from collections import OrderedDict
+from typing import Hashable, Iterable, TypeVar
+
+import torch
+from torch import Tensor
+
+from torchjd.aggregation import Aggregator
+
+from ._utils import _OrderedSet, ordered_set
+from .base import Transform
+from .tensor_dict import EmptyTensorDict, Gradients, GradientVectors, JacobianMatrices, Jacobians
+
+_KeyType = TypeVar("_KeyType", bound=Hashable)
+_ValueType = TypeVar("_ValueType")
+
+
+class Aggregate(Transform[Jacobians, Gradients]):
+    def __init__(self, aggregator: Aggregator, key_order: Iterable[Tensor]):
+        matrixify = _Matrixify(key_order)
+        strategy = _UnifyingStrategy(aggregator, key_order)
+        reshape = _Reshape(key_order)
+
+        self._aggregator_str = str(aggregator)
+        self.transform = reshape << strategy << matrixify
+
+    def _compute(self, input: Jacobians) -> Gradients:
+        return self.transform(input)
+
+    def __str__(self) -> str:
+        raise f"Aggregate {self._aggregator_str}"
+
+    @property
+    def required_keys(self) -> set[Tensor]:
+        return self.transform.required_keys
+
+    @property
+    def output_keys(self) -> set[Tensor]:
+        return self.transform.output_keys
+
+
+class _UnifyingStrategy(Transform[JacobianMatrices, GradientVectors]):
```

Comment:
This is now a weird name, I would basically make this `Aggregate` and the other one `_Aggregate`.

Reply:
I think `_AggregateMatrices` is better. It disambiguates with `Aggregate`.